Implement SSH key file path env substitution

hifi opened the pull request:
People usually want their SSH key file path to be relative to `$HOME` which helps transferring database between multiple systems with different local users. Automatic path pickup will use an absolute path but you can manually change it to use a tilde or any combination of environment variables.

For example the following key file paths work on Linux/Mac/BSD:
- `~/.ssh/id_rsa`
- `$HOME/.ssh/id_rsa`

Following hypothetical path should work on Windows:
- `%HOMEDRIVE%%HOMEPATH%\.ssh\id_rsa`

This work has been kindly supported by my employer, Vincit.

## Type of change
[NOTE]: # ( Please remove all lines which don't apply. )
- ✅ New feature (non-breaking change which adds functionality)

## Description and Context
Fixes #3523

## Testing strategy
[NOTE]: # ( Please describe in detail how you tested your changes. )
[TIP]:  # ( We expect new code to be covered by unit tests and documented with doc blocks! )
Tests are implemented for env substition. Manually tested with the GUI on Linux.

## Checklist:
[NOTE]: # ( Please go over all the following points. )
[NOTE]: # ( Again, remove any lines which don't apply. )
[NOTE]: # ( Pull Requests that don't fulfill all [REQUIRED] requisites are likely )
[NOTE]: # ( to be sent back to you for correction or will be rejected.  )
- ✅ I have read the **CONTRIBUTING** document. **[REQUIRED]**
- ✅ My code follows the code style of this project. **[REQUIRED]**
- ✅ All new and existing tests passed. **[REQUIRED]**
- ✅ I have added tests to cover my changes.

schlimmchen commented:
Could you maybe move the function that expands the tilde and environment variables in a path string to a shared source file? I wanted exactly this functionality for KeeShare path settings.

hifi commented:
Moved `KeeAgentSettings::envSubst()` to `Tools::envSubst()` for more generic use as per request. This also removed an extra test step as the tools already had a test runner. Rebased on top of current develop.

droidmonkey commented:
I rebased this onto the agent refactor.
